packages/library/src/decode-transaction.ts

```diff
@@ -0,0 +1,75 @@
+import {
+    assertAccountsDecoded,
+    assertAccountsExist,
+    type FetchAccountsConfig,
+    fetchJsonParsedAccounts,
+} from '@solana/accounts';
+import { Address } from '@solana/addresses';
+import type { GetMultipleAccountsApi } from '@solana/rpc-core';
+import type { Rpc } from '@solana/rpc-transport';
+import {
+    type AddressesByLookupTableAddress,
+    type CompilableTransaction,
+    decompileTransaction,
+    getCompiledTransactionDecoder,
+    type ITransactionWithSignatures,
+} from '@solana/transactions';
+
+let compiledTransactionDecoder: ReturnType<typeof getCompiledTransactionDecoder> | undefined = undefined;
+
+type FetchedAddressLookup = {
+    addresses: Address[];
+};
+
+async function fetchLookupTables(
+    lookupTableAddresses: Address[],
+    rpc: Rpc<GetMultipleAccountsApi>,
+    config?: FetchAccountsConfig,
+): Promise<AddressesByLookupTableAddress> {
+    const fetchedLookupTables = await fetchJsonParsedAccounts<FetchedAddressLookup[]>(
```

Comment:
Heh, to use the generated Address Lookup Table client here, or `jsonParsed`?? In the future, that is.

Reply:
Probably the generated client because it'll have better typing and we won't need that `FetchedAddressLookup` type param.